bin/update

```diff
@@ -44,28 +48,54 @@ else
     repo="$1"
 fi
 
+case "${repo}" in
+    "all"|"__all"|"crucible"|"__crucible")
+        # the first and second time this script is called for the
+        # "all" or "crucible" targets a special case is invoked to
+        # ensure that updates can be done to the crucible repository
+        # in a manner which does not break the update process
+
+        pushd $CRUCIBLE_HOME > /dev/null
+
+        # create and execute a temporary script so that changes to the
+        # "real" script do not break the updating process while it is
+        # running -- ie. handle bash does not behave when the script
+        # is changed while it is running
+        TMP_SCRIPT=$(mktemp)
+        cp -a $CRUCIBLE_HOME/bin/_update-git ${TMP_SCRIPT}
+        echo "rm ${TMP_SCRIPT}" >> ${TMP_SCRIPT}
+        echo "exec ${CRUCIBLE_HOME}/bin/update __${repo}" >> ${TMP_SCRIPT}
+
+        exec ${TMP_SCRIPT}
+        ;;
+esac
+
+# beyond this point the script behaves "normally", meaning that it
+# does not have to create a copy of itself anymore and exec that copy
 
 case "${repo}" in
-    "all"|"crucible"|"controller-image")
-        # nothing to do here yet
-        foo=1
+    "____all"|"____crucible")
+        # any crucible specific post update logic goes here
```

Comment:
this could be : as well

Reply:
Not sure what you mean about the syntax.  Logic wise, since we know we want to do some post update file migration very soon I went ahead and noted where that should be done.

Reply:
Sorry, I just meant using the built-in no-op ':' to "do nothing". Something like:
    "____all"|"____crucible")
        : echo noop